File: tooling/cloudformation/scripts/test_aws_sdk.go
package main

import (
	"context"
	"fmt"
	"log"

	"github.com/aws/aws-sdk-go-v2/config"
	"github.com/aws/aws-sdk-go-v2/service/sts"
)

func main() {
	// Load AWS configuration using container credentials
	cfg, err := config.LoadDefaultConfig(context.TODO())
	if err != nil {
		log.Fatalf("❌ Go AWS SDK test failed: unable to load SDK config, %v", err)
	}

	// Create STS client
	stsClient := sts.NewFromConfig(cfg)

	// Get caller identity
	resp, err := stsClient.GetCallerIdentity(
		context.TODO(), &sts.GetCallerIdentityInput{},
	)
	if err != nil {
		log.Fatalf("❌ Go AWS SDK test failed: %v", err)
	}

	fmt.Println("✅ Go AWS SDK test successful!")
	if resp.Account != nil {
		fmt.Printf("Account: %s\n", *resp.Account)
	}
	if resp.UserId != nil {
		fmt.Printf("User ID: %s\n", *resp.UserId)
	}
	if resp.Arn != nil {
		fmt.Printf("ARN: %s\n", *resp.Arn)
	}
}

File: tooling/cloudformation/scripts/test_aws_sdk.js
#!/usr/bin/env node
/**
 * Test script for Node.js AWS SDK with CredProxy
 */

const { STSClient, GetCallerIdentityCommand } = require('@aws-sdk/client-sts');

async function testAwsSdkCredentials() {
    try {
        // Create STS client using container credentials
        const stsClient = new STSClient({});

        // Get caller identity
        const command = new GetCallerIdentityCommand({});
        const response = await stsClient.send(command);

        console.log('✅ Node.js AWS SDK test successful!');
        console.log(`Account: ${response.Account || 'Unknown'}`);
        console.log(`User ID: ${response.UserId || 'Unknown'}`);
        console.log(`ARN: ${response.Arn || 'Unknown'}`);

        return true;

    } catch (error) {
        console.error(`❌ Node.js AWS SDK test failed: ${error.message}`);
        return false;
    }
}

testAwsSdkCredentials().then(success => {
    process.exit(success ? 0 : 1);
});
